#! /bin/sh /usr/share/dpatch/dpatch-run
## u02-lp-integration.dpatch by Felix Geyer <debfx-pkg@fobos.de>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Add "Report a problem" menu entry

@DPATCH@

diff -Nur virtualbox-ose-3.0.0-dfsg/src/VBox/Frontends/VirtualBox/src/VBoxHelpActions.h virtualbox-ose-3.0.0-dfsg.patch/src/VBox/Frontends/VirtualBox/src/VBoxHelpActions.h
--- virtualbox-ose-3.0.0-dfsg/src/VBox/Frontends/VirtualBox/src/VBoxHelpActions.h	2009-07-01 09:47:13.000000000 +0200
+++ virtualbox-ose-3.0.0-dfsg.patch/src/VBox/Frontends/VirtualBox/src/VBoxHelpActions.h	2009-07-09 17:22:27.477635065 +0200
@@ -44,7 +44,7 @@
 struct VBoxHelpActions
 {
     VBoxHelpActions()
-        : contentsAction (NULL), webAction (NULL)
+        : contentsAction (NULL), webAction (NULL), reportBugAction (NULL)
         , resetMessagesAction (NULL), registerAction (NULL)
         , updateAction (NULL), aboutAction (NULL)
         {}
@@ -55,6 +55,7 @@
 
     QAction *contentsAction;
     QAction *webAction;
+    QAction *reportBugAction;
     QAction *resetMessagesAction;
     QAction *registerAction;
     QAction *updateAction;
diff -Nur virtualbox-ose-3.0.0-dfsg/src/VBox/Frontends/VirtualBox/src/globals/VBoxProblemReporter.h virtualbox-ose-3.0.0-dfsg.patch/src/VBox/Frontends/VirtualBox/src/globals/VBoxProblemReporter.h
--- virtualbox-ose-3.0.0-dfsg/src/VBox/Frontends/VirtualBox/src/globals/VBoxProblemReporter.h	2009-07-01 09:47:13.000000000 +0200
+++ virtualbox-ose-3.0.0-dfsg.patch/src/VBox/Frontends/VirtualBox/src/globals/VBoxProblemReporter.h	2009-07-09 17:43:50.462134792 +0200
@@ -383,6 +383,7 @@
 public slots:
 
     void showHelpWebDialog();
+    void reportBugLP();
     void showHelpAboutDialog();
     void showHelpHelpDialog();
     void resetSuppressedMessages();
diff -Nur virtualbox-ose-3.0.0-dfsg/src/VBox/Frontends/VirtualBox/src/VBoxHelpActions.cpp virtualbox-ose-3.0.0-dfsg.patch/src/VBox/Frontends/VirtualBox/src/VBoxHelpActions.cpp
--- virtualbox-ose-3.0.0-dfsg/src/VBox/Frontends/VirtualBox/src/VBoxHelpActions.cpp	2009-07-01 09:47:13.000000000 +0200
+++ virtualbox-ose-3.0.0-dfsg.patch/src/VBox/Frontends/VirtualBox/src/VBoxHelpActions.cpp	2009-07-09 17:29:19.804635840 +0200
@@ -37,6 +37,9 @@
     webAction = new QAction (aParent);
     webAction->setIcon (VBoxGlobal::iconSet (":/site_16px.png"));
 
+    reportBugAction = new QAction (aParent);
+    reportBugAction->setIcon (VBoxGlobal::iconSet (":/report_bug_16px.xpm"));
+
     resetMessagesAction = new QAction (aParent);
     resetMessagesAction->setIcon (VBoxGlobal::iconSet (":/reset_16px.png"));
 
@@ -54,6 +57,8 @@
                       &vboxProblem(), SLOT (showHelpHelpDialog()));
     QObject::connect (webAction, SIGNAL (triggered()),
                       &vboxProblem(), SLOT (showHelpWebDialog()));
+    QObject::connect (reportBugAction, SIGNAL (triggered()),
+                      &vboxProblem(), SLOT (reportBugLP()));
     QObject::connect (resetMessagesAction, SIGNAL (triggered()),
                       &vboxProblem(), SLOT (resetSuppressedMessages()));
     QObject::connect (registerAction, SIGNAL (triggered()),
@@ -75,6 +80,7 @@
 
     aMenu->addAction (contentsAction);
     aMenu->addAction (webAction);
+    aMenu->addAction (reportBugAction);
     aMenu->addSeparator();
 
     aMenu->addAction (resetMessagesAction);
@@ -109,6 +115,8 @@
     webAction->setStatusTip (VBoxProblemReporter::tr (
         "Open the browser and go to the VirtualBox product web site"));
 
+    reportBugAction->setText (VBoxProblemReporter::tr ("Re&port a problem..."));
+
     resetMessagesAction->setText (VBoxProblemReporter::tr ("&Reset All Warnings"));
     resetMessagesAction->setStatusTip (VBoxProblemReporter::tr (
         "Cause all suppressed warnings and messages to be shown again"));
diff -Nur virtualbox-ose-3.0.0-dfsg/src/VBox/Frontends/VirtualBox/src/globals/VBoxProblemReporter.cpp virtualbox-ose-3.0.0-dfsg.patch/src/VBox/Frontends/VirtualBox/src/globals/VBoxProblemReporter.cpp
--- virtualbox-ose-3.0.0-dfsg/src/VBox/Frontends/VirtualBox/src/globals/VBoxProblemReporter.cpp	2009-07-01 09:47:13.000000000 +0200
+++ virtualbox-ose-3.0.0-dfsg.patch/src/VBox/Frontends/VirtualBox/src/globals/VBoxProblemReporter.cpp	2009-07-09 17:46:34.461454809 +0200
@@ -2357,6 +2357,11 @@
     vboxGlobal().openURL ("http://www.virtualbox.org");
 }
 
+void VBoxProblemReporter::reportBugLP()
+{
+    QProcess::startDetached ("launchpad-integration -P virtualbox-ose-qt -b");
+}
+
 void VBoxProblemReporter::showHelpAboutDialog()
 {
     CVirtualBox vbox = vboxGlobal().virtualBox();
diff -Nur virtualbox-ose-3.0.0-dfsg/src/VBox/Frontends/VirtualBox/VirtualBox1.qrc virtualbox-ose-3.0.0-dfsg.patch/src/VBox/Frontends/VirtualBox/VirtualBox1.qrc
--- virtualbox-ose-3.0.0-dfsg/src/VBox/Frontends/VirtualBox/VirtualBox1.qrc	2009-07-01 09:47:13.000000000 +0200
+++ virtualbox-ose-3.0.0-dfsg.patch/src/VBox/Frontends/VirtualBox/VirtualBox1.qrc	2009-07-09 17:40:48.140636318 +0200
@@ -39,6 +39,7 @@
     <file alias="site_16px.png">images/site_16px.png</file>
     <file alias="site_32px.png">images/site_32px.png</file>
     <file alias="site_disabled_32px.png">images/site_disabled_32px.png</file>
+    <file alias="report_bug_16px.xpm">../../../../debian/lpi-bug.xpm</file>
     <file alias="register_16px.png">images/register_16px.png</file>
     <file alias="register_32px.png">images/register_32px.png</file>
     <file alias="register_disabled_16px.png">images/register_disabled_16px.png</file>
